src/routes/news/containers/WithCustomNewsProvider.js

```diff
@@ -0,0 +1,357 @@
+    onItemPress(tab.type)
+  }
+  const isLocal = tab.type === LOCAL
+  const isSelected = tab.type === selectedNewsType
+  return (
+    <TouchableWrapper onPress={onPress}>
+      {isLocal ? (
+        <LocalTabWrapper isSelected={isSelected} theme={theme}>
+          <LocalText theme={theme}>{t(tab.type)}</LocalText>
+        </LocalTabWrapper>
+      ) : (
+        <NewsTypeIcon source={isSelected ? tab.active : tab.inactive} />
+      )}
+    </TouchableWrapper>
+  )
+}
+
+const TranslatedNewsTypeItem = withTranslation('news')(
+  withTheme()(NewsTypeItem)
+)
+
+const withCustomNewsProvider = <
+  S: {
+    ...ProviderPropType
+  },
+  R: {}
+>(
+    refresh: (refreshProps: R, dispatch: Dispatch<StoreActionType>) => void
+  ): ((
+  Component: React.ComponentType<S>
+) => React.ComponentType<PropsType<S, R>>) => {
+  return (
+    Component: React.ComponentType<S>
+  ): React.ComponentType<PropsType<S, R>> => {
+    return class extends React.Component<
+      PropsType<S, R>,
+      {| selectedNewsType: string |}
+    > {
+      static displayName = wrapDisplayName(Component, 'withCustomNewsProvider');
+
+      state = {
+        selectedNewsType: this.getAvailableNewsType()
+      };
+
+      listRef = null;
+
+      getAvailableNewsType (): string {
+        const { cityModel, selectedNewsType } = this.props.innerProps || {}
+        const { tunewsEnabled, pushNotificationsEnabled } = cityModel || {}
+        let type = LOCAL
+        if (selectedNewsType) {
+          type = selectedNewsType
+        } else if (tunewsEnabled && !pushNotificationsEnabled) {
+          type = TUNEWS
+        } else if (pushNotificationsEnabled && !tunewsEnabled) {
+          type = LOCAL
+        }
+        return type
+      }
+
+      selectNewsType = type => {
+        this.setState(
+          {
+            selectedNewsType: type
+          },
+          this.fetchNews
+        )
+      };
+
+      selectNewsItemAndScrollToTop = type => {
+        this.selectNewsType(type)
+      };
+
+      fetchNews = () => {
+        const { selectedNewsType } = this.state
+        const { dispatch } = this.props
+        if (this.props.innerProps) {
+          const {
+            cityCode,
+            navigation,
+            path,
+            language
+          } = this.props.innerProps
+          const fetchNews: FetchNewsActionType = {
+            type: 'FETCH_NEWS',
+            params: {
+              city: cityCode,
+              language,
+              path,
+              type: selectedNewsType,
+              key: navigation.state.key,
+              criterion: {
+                forceUpdate: false,
+                shouldRefreshResources: false
+              }
+            }
+          }
+          dispatch(fetchNews)
+        }
+      };
+
+      componentDidUpdate (prevProps: PropsType<S, R>) {
+        const prevLanguage = (prevProps.innerProps || {}).language
+        const currentLanguage = (this.props.innerProps || {}).language
```

Comment:
`currentLanguage && prevLanguage` ??

Reply:
if current language is not-undefind and ( prevLanguage != currentLanguage )